src/home/homework-luks.c

```diff
@@ -1294,22 +1327,44 @@ int home_deactivate_luks(UserRecord *h) {
         r = crypt_init_by_name(&cd, dm_name);
         if (IN_SET(r, -ENODEV, -EINVAL, -ENOENT)) {
                 log_debug_errno(r, "LUKS device %s is already detached.", dm_name);
-                return false;
+                detached = false;
         } else if (r < 0)
                 return log_error_errno(r, "Failed to initialize cryptsetup context for %s: %m", dm_name);
+        else {
+                log_info("Discovered used LUKS device %s.", dm_node);
+
+                crypt_set_log_callback(cd, cryptsetup_log_glue, NULL);
+
+                r = crypt_deactivate(cd, dm_name);
+                if (IN_SET(r, -ENODEV, -EINVAL, -ENOENT)) {
+                        log_debug_errno(r, "LUKS device %s is already detached.", dm_node);
+                        detached = false;
```

Comment:
Maybe say "was already detached" to clarify that this was in the past.
And rename `detached` to `we_detached` or something. Otherwise a comment saying that it was detatched and setting the variable to false look like a mistake.

Reply:
I'll change it to "has already been detached", after all it was done n the past, but still applies